### PR TITLE
azure-cli: patch to build with pyyaml6

### PR DESCRIPTION
**Pull request by chenrui333**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

relates to:
- https://github.com/Azure/azure-cli/pull/26919
- https://github.com/Homebrew/homebrew-core/pull/136514

### CODE REVIEW

**Comment by SMillerDev** [comment]
This should be upstreamed

**Comment by chenrui333** [comment]
> * {CI} Bump pyyaml to latest version 6.0.1 Azure/azure-cli#26919

it is already in the upstream, they just need to cut a new release (the patch does not get applied cleanly)

- https://github.com/Azure/azure-cli/pull/26919
- https://github.com/Azure/azure-cli/pull/26917

**Comment by SMillerDev** [comment]
Did that change also apply to the template that upstream uses to push formula updates?

**Comment by chenrui333** [comment]
> Did that change also apply to the template that upstream uses to push formula updates?

yeah, it would be on the template (that is also the reason why I did not add pyyaml ad dependency, but rather just a resource)

**Review by branchv — APPROVED** [review]

**Review by SMillerDev — APPROVED** [review]

**Comment by chenrui333** [comment]
```
time="2023-07-26T09:34:55Z" level=fatal msg="Uploading manifest list failed, attempted the following formats: application/vnd.oci.image.index.v1+json(uploading manifest 2.50.0_1 to ghcr.io/homebrew/core/azure-cli: received unexpected HTTP status: 502 Bad Gateway)"
```

**Review by SMillerDev — APPROVED** [review]

File: Formula/azure-cli.rb [hunk 1]
```diff
@@ -784,3 +791,18 @@ def install
     assert_equal azure_cloud["endpoints"]["resourceManager"], "https://management.azure.com/"
   end
 end
+
+__END__
+diff --git a/tools/setup.py b/tools/setup.py
```
Comment (jiasli) [inline Formula/azure-cli.rb:796]: `tools/setup.py` is actually not used as part of Azure CLI installation at all, so this patch is unnecessary.

**Pull request merged**